### ISSUE TITLE
Gaps in tabbed container

### ISSUE THREAD

**Issue by ton** [Airblader/i3#37]
Is it possible to set a gap size between containers in a tabbed container?

**Comment by Airblader** [comment]
I'm not really sure what you mean. How would there be gaps "between" contains in a tabbed container? They're not shown at the same time (that's the purpose of it).

**Comment by ton** [comment]
For example see the screenshot below, it contains one big tabbed container. I have a global gap size of 5 pixels. I would expect to have gaps between the windows in the tabbed container as well.

![tabbed](https://cloud.githubusercontent.com/assets/157002/6266955/fabd401a-b83f-11e4-95f7-1b7d0dd8f507.png)

**Comment by Airblader** [comment]
Ah, gotcha. When I find the time (no garantuees), I can take a look and see how complicated this is. My guess is, though, that this won't be too easy since there's a lot of corner cases to be considered.

**Comment by Airblader** [comment]
I was thinking whether this is even a good idea. Then something came to my mind: I haven't tried it, but I would guess that using gaps there causes the same glitch as the normal gaps regarding titlebars. I'd really rather not lose titlebars in stacked / tabbed containers as they are definitely essential there.

So this would probably require the glitch to be fixed, which I don't really see happening anytime soon. It likely requires quite a change to the code will will also affect maintainability a lot.

We'll see, though. I'll still try to at least give this a shot sometime and see what happens.

**Comment by ton** [comment]
Cool, thanks!

I think it is at least consistent to have gaps in tabbed containers as well, and of course it looks good as well ;)

**Comment by Airblader** [comment]
I took a look at this and unfortunately, my suspicion turned out to be true: it will cause the same glitch as for normal containers if they have titlebars. Since titlebars are essential for tabbed and stacked containers, this fix will require a fix for the main problem.

Unfortunately, that's unlikely. i3's way of reparenting is just not made for that and even if I managed to change that, it would massively change the way i3 works which would make maintaining it really, really hard.

So I'm sorry, but I'll close this as "won't fix". I hope you understand.
